src/Geom_Poly.f90

```diff
@@ -362,19 +374,140 @@ ELEMENTAL FUNCTION isCircle_PolygonType(thisPoly) RESULT(bool)
 !
 !-------------------------------------------------------------------------------
 !> @brief This routine will query a polygon type, check if it meets the criteria
-!>        to be a circle, and if so it will return the radius for the circle.
+!>        to be a circle or circle section, and if so it will return the radius.
 !> @param thisPoly The polygon type from which to get the radius
-!> @param r The radius if thisPoly is a circle.  It will be 0.0 for all other
-!>        cases.
+!> @param r The radius if thisPoly is a circle or circle section. It will be 0.0
+!>         for all other cases.
 !>
     ELEMENTAL FUNCTION getRadius_PolygonType(thisPoly) RESULT(r)
       CLASS(PolygonType),INTENT(IN) :: thisPoly
       REAL(SRK) :: r
       r=0.0_SRK
-      IF(thisPoly%isCircle()) r=thisPoly%quadEdge(3,1)
+      IF(thisPoly%isSection()) r=thisPoly%quadEdge(3,1)
     ENDFUNCTION getRadius_PolygonType
 !
 !-------------------------------------------------------------------------------
+!> @brief Determines whether a polygon meets the criteria for being an arc
+!>        sector. In short, the polygon must have exactly two quadratic edges
+!>        and exactly two non-quadratic edges. A circle is not an arc sector.
+!> @param thisPoly The polygon type to query
+!> @param bool The logical result of the operation
+!>
+    ELEMENTAL FUNCTION isSector_PolygonType(thisPoly) RESULT(bool)
+      CLASS(PolygonType),INTENT(IN) :: thisPoly
+      LOGICAL(SBK) :: bool
+      INTEGER(SIK) :: nNonQuadEdges!,i
+      ! REAL(SRK) :: dist,dist_old
+      bool=.FALSE.
+      IF(thisPoly%isinit) THEN
+        !nVert should be equal to the total number of edges
+        nNonQuadEdges=thisPoly%nVert-thisPoly%nQuadEdge
+        bool=(thisPoly%nQuadEdge == 2) .AND. (nNonQuadEdges == 2)
+        !The following code can be used to check if the lengths of the
+        !non-quadratic edges are equal. This might become relevant if the
+        !sector is required to be "ideal". Leaving commented now for speed.
+        ! dist_old=-1.0_SRK
+        ! IF(ALLOCATED(thisPoly%quad2edge)) THEN
+        !   DO i=1,thisPoly%nVert
+        !     IF(.NOT.ANY(thisPoly%quad2edge == i)) THEN
+        !       dist=Distance(thisPoly%vert(thisPoly%edge(1,i)), &
+        !         thisPoly%vert(thisPoly%edge(2,i)))
+        !       IF(dist_old > 0.0_SRK) THEN
+        !         bool=bool .AND. (dist .APPROXEQA. dist_old)
+        !       ELSE
+        !         dist_old=dist
+        !       ENDIF
+        !     ENDIF
+        !   ENDDO
+        ! ENDIF
+      ENDIF
+    ENDFUNCTION isSector_PolygonType
+!
+!-------------------------------------------------------------------------------
+!> @brief Determines whether a polygon meets the criteria for being an circle
+!>        section. In short, the polygon must have exactly 1 quadratic edge
+!>        and exactly two non-quadratic edges. A full circle is a section.
+!> @param thisPoly The polygon type to query
```

Comment:
Change the comment to be "two non-quadratic edges, or be a full circle".  A little clearer.

Reply:
Done.